packages/docusaurus-1.x/lib/static/js/scrollSpy.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+(function scrollSpy() {
+  const OFFSET = 10;
+  let timer;
+  let headingsCache;
+  const findHeadings = () =>
+    headingsCache || document.querySelectorAll('.toc-headings > li > a');
+  const onScroll = () => {
+    if (timer) {
+      // throttle
+      return;
+    }
+    timer = setTimeout(() => {
+      timer = null;
+      let found = false;
+      const headings = findHeadings();
+      for (let i = 0; i < headings.length; i++) {
+        // if !found and i is the last element, highlight the last
+        let current = !found;
+        if (!found && i < headings.length - 1) {
+          const next = headings[i + 1].href.split('#')[1];
+          const nextHeader = document.getElementById(next);
+          const top = nextHeader.getBoundingClientRect().top;
+          // The following tests whether top + scrollTop
+          // (the top of the header) is greater than scrollTop
+          // (where scrollTop = window.pageYOffset, the top of
+          // the window), with OFFSET pixels of slop.
+          current = top > OFFSET;
+        }
+        if (current) {
+          found = true;
+          headings[i].className = 'active';
+        } else {
+          headings[i].className = '';
```

Comment:
This erases all class names on the element and will break if we add more classes in future.

```js
headings[i].classList.remove('active');
```